core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -285,6 +287,20 @@ trait WorkflowComponent {
       groupedSeq.map(_.toMap)
     }
 
+    def countWorkflowsByQueueStatusByUser: ReadAction[Map[String, Map[String, Int]]] = {
+      // Query for a Seq of (User, Workflow) tuples.
+      val userWorkflows = (for {
+        workflow <- findQueuedAndRunningWorkflows
+        submission <- submissionQuery if workflow.submissionId === submission.id
+        user <- rawlsUserQuery if submission.submitterId === user.userSubjectId
+      } yield (user, workflow)).result
+
+      // Combine results into a nested map using Scalaz semigroups for aggregation.
```

Comment:
Heh, I was trying to gauge people's reactions by importing scalaz. :)

I feel like a lot of FP ideas have confusing names, but are actually relatively simple concepts. In particular a semigroup abstracts _combining_ things of the same type. Scalaz provides implicit semigroups for several different types, which can all be invoked with the `|+|` operator. 

For example, the semigroup for `Int` is addition:
```
scala> import scalaz._, Scalaz._
import scalaz._
import Scalaz._

scala> 1 |+| 2
res0: Int = 3
```
The semigroup for `List` is concatenation:
```
scala> List(1,2,3) |+| List(4,5,6)
res1: List[Int] = List(1, 2, 3, 4, 5, 6)
```
The semigroup for `Map` groups elements of the same key, and applies the semigroup for the value type:
```
scala> Map('a' -> 1, 'b' -> 2) |+| Map('a' -> 42, 'c' -> 3)
res2: scala.collection.immutable.Map[Char,Int] = Map(a -> 43, c -> 3, b -> 2)
```

You get the idea. So in this case we're using the semigroups for `Map` and `Int` to reduce a list of tuples into a `Map[String, Map[String, Int]]`. 

I could write it without scalaz using standard library functions, but I would argue it would make the code _more_ complicated. :) That said, I can give it a shot.

Also, as I was discussing with @helgridly, I'd be happy to give workbench folks an overview of some FP concepts, like semigroups, monoids, functors, monads, etc if people are interested. :)